### PR TITLE
RUST-1987 Add serde helper module for `Option<DateTime>`

### PR DESCRIPTION
**Pull request by lazureykis**

### CODE REVIEW

**Comment by isabelatkinson** [comment]
Hi @lazureykis, thanks for opening this PR! Can you please add some tests for this functionality to [this file](https://github.com/mongodb/bson-rust/blob/main/src/tests/serde.rs)?

File: src/serde_helpers.rs [hunk 1]
```diff
@@ -307,6 +313,50 @@ pub mod chrono_datetime_as_bson_datetime {
     }
 }
 
+/// Contains functions to serialize a [`Option<chrono::DateTime>`] as a [`Option<crate::DateTime>`] and deserialize
```
Comment (isabelatkinson) [inline src/serde_helpers.rs:316]:
```suggestion
/// Contains functions to serialize an [`Option<chrono::DateTime>`] as an [`Option<crate::DateTime>`] and deserialize
```

File: src/serde_helpers.rs [hunk 2]
```diff
@@ -307,6 +313,50 @@ pub mod chrono_datetime_as_bson_datetime {
...
+/// a [`Option<chrono::DateTime>`] from a [`Option<crate::DateTime>`].
```
Comment (isabelatkinson) [inline src/serde_helpers.rs:317]:
```suggestion
/// an [`Option<chrono::DateTime>`] from an [`Option<crate::DateTime>`].
```

**Review by isabelatkinson — CHANGES_REQUESTED** [review]
Looks like there are some formatting errors. You can run the `check-rustfmt.sh` script in the `.evergreen` directory to determine where the errors are coming from and then use `rustfmt +nightly --unstable-features <filename>` to fix them.

**Comment by lazureykis** [comment]
> Looks like there are some formatting errors. You can run the `check-rustfmt.sh` script in the `.evergreen` directory to determine where the errors are coming from and then use `rustfmt +nightly --unstable-features <filename>` to fix them.

Thanks. Should be fixed now.

**Review by isabelatkinson — APPROVED** [review]
looks great - going to tag in the other team member for approval and then we can merge this in! thanks for the contribution!

**Pull request merged**